openmls/src/treesync/diff.rs

```diff
@@ -252,17 +252,18 @@ impl<'a> TreeSyncDiff<'a> {
         Ok(())
     }
 
-    /// Derive a new direct path for our own leaf from the given `leaf_secret`.
+    /// Derive a new direct path for our own leaf
     ///
     /// Returns an error if the own leaf is not in the tree
-    fn derive_path_from_leaf_secret(
+    fn derive_path(
         &self,
         backend: &impl OpenMlsCryptoProvider,
         ciphersuite: Ciphersuite,
-        leaf_secret: Secret,
     ) -> Result<PathDerivationResult, LibraryError> {
-        let leaf_path_secret = PathSecret::from(leaf_secret);
-        let path_secret = leaf_path_secret.derive_path_secret(backend, ciphersuite)?;
+        let path_secret = PathSecret::from(
+            Secret::random(ciphersuite, backend, None)
```

Comment:
Should this be versioned?

Reply:
I think at this point, there is no need. There is currently no clear story for how we handle version agility. That should be tackled with #1065.